src/main/java/org/jruby/ext/openssl/OpenSSL.java

```diff
@@ -255,27 +255,27 @@ static void warn(final ThreadContext context, final IRubyObject msg) {
     public static String javaVersion(final String def) {
         final String javaVersionProperty =
                 SafePropertyAccessor.getProperty("java.version", def);
-        if ( javaVersionProperty == "0" ) return "1.7.0"; // Android
+        if (javaVersionProperty.equals("0")) return "1.7.0"; // Android
```

Comment:
this change makes it no longer null-safe as it was before - could you please swap the equals comparison, just in case